aiosmtpd/smtp.py

```diff
@@ -64,7 +65,10 @@ def __init__(self, handler,
         self.seen_greeting = ''
         self.extended_smtp = False
         self.command_size_limits.clear()
-        self.fqdn = socket.getfqdn()   # XXX this blocks, fix it?
+        if not hostname:
+            self.hostname = socket.getfqdn()   # XXX this blocks, fix it?
+        else:
+            self.hostname = hostname
```

Comment:
Another thought.  I generally prefer positive tests over negative tests, all things being equal.  E.g. `if hostname` instead of `if not hostname`.  I find it's easier to follow positive test logic.  But in this case, I think you can even use a ternary expression, e.g.:

```
self.hostname = (socket.getfqdn() if hostname is None else hostname)
```

I generally also like more narrow conditionals where it makes sense, e.g. if hostname isn't going to be generally false-y but really will only be None or some string value, test against None.